Migration of IDE Troubleshooting from Support Articles to central doc… (#15582)
rstudio/rstudio: docs/user/rstudio/ide/guide/troubleshooting/resetting-state.qmd

---
title: "Resetting RStudio's State"
date-meta: 2025-01-08
---

> # Overview

Reviewer: Should be an h2

> RStudio Desktop stores its internal state in a hidden directory. If this directory does not exist, RStudio will create it on start up. This directory includes information about open documents, log files, and other state information. Removing (or renaming) this directory will reset RStudio's state. 

Reviewer: ```suggestion
RStudio Desktop stores its internal state in a hidden directory. If this directory does not exist, RStudio creates it on start-up. This directory includes information about open documents, log files, and other state information. Removing (or renaming) this directory resets RStudio's state. 
```

> We recommend renaming this directory to create a backup version instead of completely deleting it. If you experience a crash or RStudio fails to start, this directory may contain vital information for determining the source of the error. In this case, we recommend renaming this directory and sending it along if asked by Posit's support team.

Reviewer: ```suggestion
Instead of deleting this directory, we recommend renaming this directory to create a backup version. If you experience a crash or RStudio fails to start, this directory contains vital information for determining the source of the error. In this case, we recommend renaming this directory and forwarding it to Posit's support team.
```

> User preferences are stored in a separate folder from internal state. This lets you perform a state reset without losing your settings, and also allows for preferences to be sync'ed between machines - for example in `%AppData%` on Windows - while internal state is machine specific. 

Reviewer: ```suggestion
User preferences are stored in a separate folder from the internal state. This allows you to perform a state reset without losing your settings. Additionally, it allows for preferences to be synced between machines. For example, in `%AppData%` on Windows, the internal state is machine-specific. 
```

> Some versions of RStudio Desktop store additional preferences (such as the size and location of the window and the rendering mode) in a separate location. To fully reset state, this must also be deleted or renamed, as described below in "Resetting Other Preferences".

Reviewer: ```suggestion
Some versions of RStudio Desktop store additional preferences (such as the size and location of the window and the rendering mode) in a separate location. To fully reset state, this must also be deleted or renamed. Learn more in the Resetting Other Preferences section below.
```

Reviewer: Add a link to the section referenced

> If you are using RStudio Projects, we'd also recommend resetting the project-specific state if you're having issues - you can do this by navigating to the Project's folder in your file browser, and renaming the `.Rproj.user` directory there.

Reviewer: ```suggestion
If you are using RStudio Projects and you are having issues, reset the project-specific state by navigating to the Project's folder in your file browser and renaming the `.Rproj.user` directory.
```

> # Accessing the RStudio-Desktop Directory (Internal State)

Reviewer: You cannot have multiple H1s ever. You should also never have an H1 on a page with a set title. 

```suggestion
## Accessing the RStudio Desktop directory (internal state)
```

## Windows

> You can open an Explorer window into the RStudio-Desktop directory by typing the following command into **Start > Run**:

Reviewer: ```suggestion
Open an Explorer window in the RStudio Desktop directory by typing the following command into **Start** > **Run**:
```

> ```powershell

Reviewer: Same comment as above

%localappdata%\RStudio
```

> You can then rename this directory and send it with your support request.

Reviewer: ```suggestion
Then, rename this directory and send it with your support request.
```

## macOS

> You can easily create a backup by running the following command from the terminal. Then you can send this backup with your support request:

Reviewer: ```suggestion
Create a backup by running the following command from the terminal. Then, send this backup with your support request:
```

```zsh
mv ~/.local/share/rstudio ~/.local/share/rstudio-backup
```

> Alternatively, you can open a Finder window into the RStudio directory by typing the following command at the Terminal:

Reviewer: ```suggestion
Alternatively, open a Finder window into the RStudio directory by typing the following command at the Terminal:
```

> ```zsh

Reviewer: same comment

open ~/.local/share/rstudio
```

## Linux

> You can easily create a backup by running the following command from the terminal. Then you can send this backup with your support request:

Reviewer: ```suggestion
Create a backup by running the following command from the terminal. Then, send this backup with your support request:
```

```bash
mv ~/.local/share/rstudio ~/.local/share/rstudio-backup
```

> Alternatively, you can open a File Browser into the RStudio directory by typing the following command at the Terminal:

Reviewer: ```suggestion
Alternatively, open a File Browser in the RStudio directory by typing the following command at the Terminal:
```

```bash
xdg-open ~/.local/share/rstudio
```

> # The RStudio Configuration Directory

Reviewer: Same comment as above

> Most problems do not require clearing preferences to resolve, but you will need to rename or remove this folder if you wish to perform a full factory reset. 

Reviewer: ```suggestion
Most problems do not require clearing preferences to resolve. However, you need to rename or remove this folder if to perform a full factory reset. 
```

## Windows

> You can open an Explorer window into the RStudio preferences directory by typing the following command into **Start > Run**:

Reviewer: ```suggestion
Open an Explorer window into the RStudio preferences directory by typing the following command into **Start** > **Run**:
```

> ```powershell

Reviewer: Same comment as above

%appdata%\RStudio
```

> You can then rename this directory to backup-rstudio and send it with your support request.

Reviewer: ```suggestion
Then, rename this directory to backup Rstudio and send it with your support request.
```

Author: The second recommendation isn't correct. The instruction is to rename a directory called `RStudio` to a directory called `backup-RStudio`. I have included the capital in the change.

## macOS

> To create a backup, run the following command from the terminal. You can then include the file `~/backup-rstudio-prefs` with your support request:

Reviewer: ```suggestion
To create a backup, run the following command from the terminal. Then, include the file `~/backup-rstudio-prefs` with your support request:
```

> ```zsh

Reviewer: Same comment as above

defaults read com.rstudio.desktop > ~/backup-rstudio-prefs
```

> To delete these settings, run the following command from the terminal. Use caution; this cannot be undone unless you have created the backup file, and deleting anything other than `com.rstudio.desktop` could create serious problems with your system.

Reviewer: ```suggestion
To delete these settings, run the following command from the terminal. 

:::{.callout-warning}
This action cannot be undone unless you have created the backup file. Deleting anything other than `com.rstudio.desktop` could create serious problems with your system.
:::

```

```zsh
defaults delete com.rstudio.desktop
```

## Linux

> You can easily create a backup by running the following command from the terminal. Then you can send this backup with your support request:

Reviewer: ```suggestion
Create a backup by running the following command from the terminal. Then, send this backup with your support request:
```

```bash
mv ~/.config/rstudio ~/backup-rstudio
```

> Alternatively, you can open a File Browser into the RStudio preferences directory by typing the following command at the Terminal:

Reviewer: ```suggestion
Alternatively, open a File Browser in the RStudio preferences directory by typing the following command at the Terminal:
```
